File: ViewModels/DataTableResponseVM.cs
using System.Collections.Generic;

namespace ViewModels
{
#pragma warning disable IDE1006 // Naming Styles
    public class DataTableResponseVM<T>
    {
        public IEnumerable<T> data { get; set; }
        public string error { get; set; }
        public int recordsFiltered { get; set; }
        public int recordsTotal { get; set; }
        public int draw { get; set; }
    }
#pragma warning restore IDE1006 // Naming Styles
}

File: ViewModelsVB/DataTableResponseVM.vb
Public Class DataTableResponseVM(Of T)
#Disable Warning IDE1006 REM Naming Styles
    Public Property data As IEnumerable(Of T)
    Public Property [error] As String
    Public Property recordsFiltered As Integer
    Public Property recordsTotal As Integer
    Public Property draw As Integer
#Enable Warning IDE1006 REM Naming Styles
End Class
